mobile/native/src/trade/order/handler.rs

```diff
@@ -17,7 +18,7 @@ pub async fn submit_order(order: Order) -> Result<()> {
     let url = format!("http://{}", config::get_http_endpoint());
     let orderbook_client = OrderbookClient::new(Url::parse(&url)?);
 
-    if let Err(e) = position::handler::update_position_after_order_submitted(order) {
+    if let Err(e) = position::handler::get_position_matching_order(&order) {
```

Comment:
I remembered another thing 😅

It was on purpose that we updated the position immediately after submitting the order in case we have a closing scenarios - this update is basically what sets the button state to `Closing` in the UI so the user does not click it again. 

I think the best way would be updating the position to `Closing` immediately in case we are in a closing scenarios, but resetting it to open in case we fail to submit the order. I think resetting the state to open was missing, not necessarily moving the update to a later point.

Sorry, I did not think about this detail yesterday.

Reply:
Best would be to change the state in the UI to set it to `Closing` upon clock until we get an update from the backend. But that might be a bit tricky as well :)

Reply:
I perceive the risk from this being small:  the action is swiping not just clicking and we get the success dialog in between.

(e.g. one can't fat-finger twice accidentally the same way as with buttons).

This is a shortcoming though, and I'll work to solve it!